### ISSUE TITLE
Reporting feature for spam comments

### ISSUE THREAD

**Issue by iambibhas** [hasgeek/funnel#520]
We need a reporting feature for comments because of increase in spam comments. This may work like this -

1. Each comment will have a `report spam` button below it along with `reply` etc, that will lead to the report creation page for logged in users, login page for anon users, forwarding them to right page after login
2. Introduce a `CommentReport` model with `reason=(<list of types of offensive contents>)`, `resolved=(true, false)`, `action=(delete_comment, suspend_user, ignore)`. 
3. Add a dashboard for reported comments, able to see the resolved and unresolved reports, with relevant actions beside them
4. each unresolved report will have an action page with above actions 
5. email/alert admins on new reports with link to the dashboard/actions page for the report
6. optionally, the page can have a list of suspended users because of spam, with option to un-suspend them, just in case of mistakenly taken actions, but this can be added only when there are frequent cases. Not needed in first release.

**Comment by jace** [comment]
Hasjob's report model may be a useful reference. It uses two models, one for the types of reports and another for the reports themselves. Since types are generally static and also have a different user flow for each, I'd skip the report type model here and only have a report model.

**Comment by iambibhas** [comment]
Types of reports in hasjob - 
```
  name   |                 title
---------+----------------------------------------
 spam    | Spam, not a job listing
 fake    | Appears to be a fake listing
 anon    | Organization is not clearly identified
 unclear | Job position is not properly described
```

**Comment by iambibhas** [comment]
There has been a surge of spam comments on funnel lately. I went through around 50 accounts and tried to figure out how they signed up. Looks like almost all of them signed up using Google auth. We have recaptcha only on registration form using email/password. So looks like spammers are using google auth to circumvent that.

**Comment by iambibhas** [comment]
By default comments are public, comments with links in them can be marked as SCREENED and put up to a moderation queue, which needs to have it's own dashboard. A dashboard with a list of all screened comments, with an option to delete comment, mark as spam(which bans the user) or ignore the report(which makes the comments PUBLIC)

**Comment by iambibhas** [comment]
All the first comment from new users can be marked as screened as well.

**Comment by jace** [comment]
1. Users get a new state column for moderation status. Values: moderated (default), regular, blocked
2. When a screened user posts a comment, it gets the existing but unused SCREENED status instead of PUBLIC status.
3. Moderation queue tool can clear or block a single comment, or user + all their comments.

**Comment by iambibhas** [comment]
In #669 we introduced comment moderation and review. Anyone can now report a spam comment. Two moderators are then asked to look at it, and if they concur, the comment is removed. If they disagree with each other, a third moderator is asked to look at it. With this mechanism, we hope there is proactive review of spam and also a safety net when comments are incorrectly reported as spam.

We also went through the entire back catalogue and reported and reviewed all spam comments.

We're not screening new users or users whose comments have been marked as spam, yet.
